package amplet2.icmp;

message Report {
	optional Header header = 1;
	repeated Item reports = 2;
}

message Header {
	optional uint32 packet_size = 1 [default = 84];
	optional bool random = 2 [default = false];
}

message Item {
	optional bytes address = 1;
	optional uint32 family = 2;
	optional sint32 rtt = 3;
	optional uint32 err_type = 4;
	optional uint32 err_code = 5;
	optional uint32 ttl = 6;
	optional string name = 7;
}
